data_prep/roberta_graph_dataset.py

```diff
@@ -49,3 +55,27 @@ def _generate_features(self):
             features_words = torch.stack(features_words).to(self._device)
 
         return features_docs, features_words
+
+    def get_doc_embedder(self):
+        """
+        Either loads a pretrained/fine-tuned roberta encoder or a non-fine-tuned one and returns it.
+        Returns:
+            encoder (RobertaModel): Roberta encoder.
+        """
+
+        if self.roberta_checkpoint is None:
+            # use only pretrained roberta for the document embeddings
+            return RobertaModel.from_pretrained('roberta-base')
+
+        # use pretrained + fine tuned roberta for the document embeddings
+        encoder_state_dict = {}
+        for layer_key, param in torch.load(self.roberta_checkpoint)['state_dict'].items():
```

Comment:
I will adapt this code once Mats branch is merged, because he is doing the exact same. Will use his method then.